sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/CombineFnsTest.java

```diff
@@ -278,6 +280,63 @@ public void testComposedCombineNullValues() {
     p.run();
   }
 
+  @Test
+  public void testComposedCombineDisplayData() {
+    SimpleFunction<String, String> extractFn = new SimpleFunction<String, String>() {
+      @Override
+      public String apply(String input) {
+        return input;
+      }
+    };
+
+    DisplayDataCombineFn combineFn1 = new DisplayDataCombineFn("combineFn1");
+    DisplayDataCombineFn combineFn2 = new DisplayDataCombineFn("combineFn2");
+
+    CombineFns.ComposedCombineFn<String> composedCombine = CombineFns.compose()
+        .with(extractFn, combineFn1, new TupleTag<String>())
+        .with(extractFn, combineFn2, new TupleTag<String>());
+
+    DisplayData displayData = DisplayData.from(composedCombine);
+    assertThat(displayData, includes(combineFn1));
+    assertThat(displayData, includes(combineFn2));
+  }
+
+  private static class DisplayDataCombineFn extends Combine.CombineFn<String, String, String> {
+    private final String value;
+    private final String key;
+    private static int i;
+
+    DisplayDataCombineFn(String value) {
+      this.key = "key" + (++i);
```

Comment:
The intent is just to have an arbitrary unique key per-instance. I've made this more clear in the latest iteration.